### PR TITLE
Automatically Link a Database to a Project

### PR DESCRIPTION
**Pull request by andreia-oca**
This PR does the following improvements:
- Automatically links a created database to a corresponding project
- Adds Delete for Database and Project (to leverage pulumi destroy during demos/testing)/
- Fix some linting warmigs
- Removes `fmt.Printf` becase they appear in Diagnostics and we don't necessarily want that

### CODE REVIEW

File: provider/resources/database.go [hunk 1]
```diff
@@ -11,15 +11,16 @@ import (
 type Database struct{}
 
 type DatabaseArgs struct {
-	Name      string  `pulumi:"name"`
-	Type      *string `pulumi:"type,optional"`
-	Region    *string `pulumi:"region,optional"`
+	ProjectName string  `pulumi:"projectName"`
```
Comment (Virgil993) [inline provider/resources/database.go:14]: Do we want to make this a required field? What if i want to create a database that isn't linked to one specific project?

File: provider/resources/database.go [hunk 2]
```diff
@@ -61,36 +72,50 @@ func (*Database) Create(ctx p.Context, name string, input DatabaseArgs, preview
 		return name, state, nil
 	}
 
-	
 	databaseType := "postgres-neon"
 	if input.Type != nil {
 		databaseType = *input.Type
 	}
-	region := "aws-us-east-1"
+	region := "us-east-1"
```
Comment (Virgil993) [inline provider/resources/database.go:79]: region for postgres database is `aws-us-east-1`

**Review by Virgil993 — COMMENTED** [review]
I think we should not make it neccesary to link a database to a project. it should be optional. I might want to create a database without creating a project

File: provider/resources/database.go (hunk 1 above)
Comment (andreia-oca) [reply provider/resources/database.go:14]: Fair point, I will make it optional

**Pull request merged**